crates/core/src/spk_client.rs

```diff
@@ -387,14 +388,14 @@ impl<I> SyncRequest<I> {
 /// See also [`SyncRequest`].
 #[must_use]
 #[derive(Debug)]
-pub struct SyncResponse<A = ConfirmationBlockTime> {
+pub struct SyncResponse<B = BlockHash, A = ConfirmationBlockTime> {
```

Comment:
I know this is a breaking change, but introducing a generic parameter before one that already exists can be really annoying. Introduce it at the end please.